File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int left = 0;
        for (auto right = 0; right < nums.size(); right++) {
            if (nums[right] != val) {
                swap(nums[left], nums[right]);
                left++;
            }
        }
        return left;
    }
};

File: 27-remove-element/remove-element.go
func removeElement(nums []int, val int) int {
    j := 0
    for i, v := range nums {
        if j == i && nums[j] != val {
            j++
        } else if nums[j] == val && v != val {
            nums[i], nums[j] = nums[j], nums[i]
            j++
        }
    }
    return j
}



    